### DISCUSSION TITLE
How to restrict files to matching dir-or-file names?

### DISCUSSION THREAD

**Discussion by douglasdd** [BurntSushi/ripgrep#1567] [General] [answered]
My use case:  In an enormous codebase, I'd like to restrict searching to only those files owned or related to a particular team/feature ("FooBar" for example).
* (a) `.../AFooBarFileName.xyz`  <-- files like this in any directory
* (b) `.../ADirNameWithFooBarInside/.../.../...`   <-- all files+dirs deeper under here

If I had bash4 with `globstar` (darned Apple!) I think I could specify the search paths like
* `**/*FooBar*/**`, and 
* `**/*FooBar*`

I can get close with:

    rg --iglob '*FooBar*' \    ## this handles case (a)
        --iglob '*FooBar*/' \  ## sorta does (b), but misses files in further nested subdirs
        'pattern'  

Does this make sense?  (If not I can try to create a .tar example test case)

**Comment by BurntSushi** [comment, marked as answer]
Seems like `rg --iglob '**/*FooBar*/**'` should work. So you were almost there. :-)

Also, I'm pretty sure zsh is available on macOS (isn't it the default nowadays?), and I believe that supports `**`.

**Reply by douglasdd** [reply]
Ah, thanks.  
((Being heavily invested in bash (& bashrc;), I didn't pay attention to zsh & thought it was only coming in future macOS versions.))
